### ISSUE TITLE
Different behavior between the Github action and the CLI

### ISSUE THREAD

**Issue by AntoniosBarotsis** [obi1kenobi/cargo-semver-checks#372]
### Steps to reproduce the bug with the above code

I was upgrading a crate from `0.1.0-rc2` to `0.1.0`. No actual code changes were present, just exiting the release candidate phase.

### Actual Behaviour

I used

```yml
      - name: Check semver
        uses: obi1kenobi/cargo-semver-checks-action@v1
        with:
          crate-name: qr-rs-lib
```

as part of my CD pipeline and got [this](https://github.com/AntoniosBarotsis/qr-rs/actions/runs/4197182627/jobs/7279161633) error.

```
Crate qr-rs-lib current version: 0.1.0
Versions on crates.io:
0.1.0-rc2
0.1.0-rc1

There is no suitable comparison version.
The current version 0.1.0 is smaller than any version published on crates.io
+ export COMPARISON_TAG=v
+ COMPARISON_TAG=v
+ git fetch --depth=1 origin +refs/tags/v:refs/tags/v
fatal: couldn't find remote ref refs/tags/v
Error: Process completed with exit code 128.
```

My workflow can be found [here](https://github.com/AntoniosBarotsis/qr-rs/blob/8d2412ebdba0c51236a67d188dc9df02c4793adc/.github/workflows/release-lib.yml) and the relevant part is [this](https://github.com/AntoniosBarotsis/qr-rs/blob/8d2412ebdba0c51236a67d188dc9df02c4793adc/.github/workflows/release-lib.yml#L17-L20).

### Expected Behaviour

This should not throw an error. The interesting thing is how the CLI does not consider this to be wrong (which is the expected behavior).

In fact, [that is how I *fixed* my pipeline, by using the CLI instead of the action](https://github.com/AntoniosBarotsis/qr-rs/commit/d66622f0400857d660fa6c9a3ae0f067aa797795)

```yml
      - name: Check semver
        working-directory: lib
        run: |
          cargo install cargo-semver-checks --locked
          cargo semver-checks check-release
```

### Generated System Information

```
#### Software version

cargo-semver-checks 0.17.1 (af71be0)

#### Operating system

Windows 6.2.9200

#### Command-line

```bash
C:\Users\anton\.cargo\bin\cargo-semver-checks.exe semver-checks --bugreport
```

#### cargo version

```
> cargo -V
cargo 1.67.1 (8ecd4f20a 2023-01-10)
```

#### Compile time information

- Profile: release
- Target triple: x86_64-pc-windows-msvc
- Family: windows
- OS: windows
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: fxsr,sse,sse2
- Host: x86_64-pc-windows-msvc
```

### Build Configuration

My `config.toml` can be found [here](https://github.com/AntoniosBarotsis/qr-rs/blob/8d2412ebdba0c51236a67d188dc9df02c4793adc/.cargo/config.toml) however, it only contains *~way too many~* lints and was present both in the failing and passing scenarios.

### Additional Context

_No response_

**Comment by AntoniosBarotsis** [comment]
The error seems to originate from [here](https://github.com/obi1kenobi/cargo-semver-checks-action/blob/29dd54c10c424222339892f0578e1c5feb5329a6/find_comparison_version.sh#L52-L57).

**Comment by AntoniosBarotsis** [comment]
It seems that the `-rc` versions take precedence over the full release which shouldn't be the case. I ran the `sh` file as if I was uploading a new `0.2.0` version and the `OUTPUT` variable was set to `0.1.0-rc2`, this should be `0.1.0` instead

```sh
CURRENT_VERSION="0.2.0"
CRATE_NAME="qr_rs_lib"

EXISTING_VERSIONS="
$( \
    curl 2>/dev/null "https://crates.io/api/v1/crates/$CRATE_NAME" | \
    jq --exit-status -r .versions[].num \
)"
echo >&2 -e "Versions on crates.io:$EXISTING_VERSIONS\n"
# Versions on crates.io:
# 0.1.0
# 0.1.0-rc2
# 0.1.0-rc1

OUTPUT="$( \
    echo -e "$CURRENT_VERSION$EXISTING_VERSIONS" | \
    sort -V | \
    grep -B 1 -A 0 --line-regexp --max-count=1 "$CURRENT_VERSION" | \
    head -n 1 \
)"

echo $OUTPUT
```
